gateway-service-impl/src/main/java/org/hypertrace/gateway/service/entity/query/ExecutionTreeBuilder.java

```diff
@@ -305,6 +316,104 @@ QueryNode buildFilterTree(EntitiesRequest entitiesRequest, Filter filter) {
     }
   }
 
+  QueryNode buildAndFilterTree(EntitiesRequest entitiesRequest) {
+    // If the filter by and order by are from QS, pagination can be pushed down to QS
+    // Since the filter and order by are from QS, there won't be any filter on other
```

Comment:
Isn't the single source fetch is already handled at the beginning of the function?